# Next delivery: `/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022`

## List of files to extract metadata

In [1]:
%%bash
dir_with_sipecam_data=/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022
list_of_files_and_subdirectories_to_extract_metadata --input_directory "$dir_with_sipecam_data"

## Check

In [2]:
%%bash
ls -lh /shared_volume/sipecam_subdirectories_02-06-2022.txt

-rw-r--r-- 1 root root 15K Jun  2 12:26 /shared_volume/sipecam_subdirectories_02-06-2022.txt


In [3]:
%%bash
ls -lh /shared_volume/sipecam_files_to_extract_metadata_from_02-06-2022.txt

-rw-r--r-- 1 root root 10M Jun  2 12:26 /shared_volume/sipecam_files_to_extract_metadata_from_02-06-2022.txt


In [4]:
%%bash
wc -l /shared_volume/sipecam_subdirectories_02-06-2022.txt

130 /shared_volume/sipecam_subdirectories_02-06-2022.txt


In [5]:
%%bash
wc -l /shared_volume/sipecam_files_to_extract_metadata_from_02-06-2022.txt

82548 /shared_volume/sipecam_files_to_extract_metadata_from_02-06-2022.txt


In [6]:
%%bash
#check every first subdirectory has a JPG, WAV or AVI file
SAVEIFS=$IFS;IFS=$(echo -en "\n\b")
for f in $(cat /shared_volume/sipecam_subdirectories_02-06-2022.txt)
  do echo "$f"
     ls "$f"|grep "[WAV|wav|JPG|jpg|AVI|avi]$"|head -n 3
done
IFS=$SAVEIFS

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Ultra/10263609
20210910_214000.WAV
20210910_215000.WAV
20210910_220000.WAV
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Grabadoras/Audible 2/10261149
19700101_000026.WAV
20210811_180000.WAV
20210811_182000.WAV
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Audible/10262012
20210821_190000.WAV
20210821_192000.WAV
20210821_194000.WAV
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoths.N5.T4
20210805_100000.WAV
20210805_101000.WAV
20210805_102000.WAV
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/Ultra.N5.T3
20210805_084432.WAV
20210805_180000.WAV
20210805_182000.WAV
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoth
20210

**Files to be checked**

"/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1392 NICTEHA 2/10262012/0~FA674D.WAV" has 0 bytes then will be moved to dir error inside latter dir

"/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1392 NICTEHA 2/10263609/memoria vacia.jpg" as is the only file in "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1392 NICTEHA 2/10263609" then latter dir will be deleted from: `/shared_volume/sipecam_subdirectories_02-06-2022.txt`

"/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1392 NICTEHA 2/10262223/memoria vacia.jpg" as is the only file in "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1392 NICTEHA 2/10262223/" then latter dir will be deleted from: `/shared_volume/sipecam_subdirectories_02-06-2022.txt`


# Which directories are of audio?


In [7]:
%%bash
if [ -f ~/list_dirs_with_audios.txt ]; then
     rm ~/list_dirs_with_audios.txt
fi
if [ -f ~/list_dirs_with_audios_auxiliary.txt ]; then
     rm ~/list_dirs_with_audios_auxiliary.txt
fi
SAVEIFS=$IFS;IFS=$(echo -en "\n\b")
for f in $(find /LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/ -name "*.WAV")
  do
     dir_audio=$(dirname $f)
     echo "$dir_audio" >> ~/list_dirs_with_audios_auxiliary.txt
done
dirs_audio=$(cat ~/list_dirs_with_audios_auxiliary.txt|uniq)
echo "$dirs_audio" > ~/list_dirs_with_audios.txt
IFS=$SAVEIFS

In [8]:
%%bash
cat ~/list_dirs_with_audios.txt

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Ultra/10263609
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Grabadoras/Audible 2/10261149
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Audible/10262012
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoths.N5.T4
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/Ultra.N5.T3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoth
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 2/AudioMoths.N5.T4/Audio
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPEC

# check_if_there_are_audio_empty_files_in_dir_and_move_them

In [9]:
%%bash
SAVEIFS=$IFS;IFS=$(echo -en "\n\b")
for dir_to_be_processed in $(cat ~/list_dirs_with_audios.txt)
    do echo "$dir_to_be_processed"
    check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10
done
IFS=$SAVEIFS

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Ultra/10263609
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Grabadoras/Audible 2/10261149
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Audible/10262012
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoths.N5.T4
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/Ultra.N5.T3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoth
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 2/AudioMoths.N5.T4/Audio
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPEC

# Extract serial number, dates and metadata of files and device

In [1]:
%%bash
list_of_dirs_to_be_processed=/shared_volume/sipecam_subdirectories_02-06-2022.txt
SAVEIFS=$IFS;IFS=$(echo -en "\n\b")
for dir_to_be_processed in $(cat $list_of_dirs_to_be_processed)
    do
    echo "$dir_to_be_processed"
    extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed" --parallel --number_of_processes 10
done
IFS=$SAVEIFS

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Ultra/10263609
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Grabadoras/Audible 2/10261149
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Audible/10262012
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoths.N5.T4
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/Ultra.N5.T3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoth
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 2/AudioMoths.N5.T4/Audio
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPEC

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1394 NICTEHA 1/10262223


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1394 NICTEHA 1/10262089
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1394 NICTEHA 1/10262039
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/found.000/dir0000.chk/HLPXGM 09048480


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/2/Grabadoras/Audible/10261149
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609


multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 147, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 91, in get_date
    match_date = ONLY_DATE_REGEX.search(comment)
TypeError: expected string or bytes-like object
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/extra

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10262223


multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 147, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 91, in get_date
    match_date = ONLY_DATE_REGEX.search(comment)
TypeError: expected string or bytes-like object
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/extra

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/GRABADORAS/4
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/GRABADORAS/2


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/GRABADORAS/1


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/GRABADORAS/5
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/GRABADORAS/3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/found.000/dir0000.chk/HLPXGM 09048480/DCIM/100RECNX


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 254, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 136, in extract_metadata_of_files
    serial_number = d_output["MetadataDevice"]["SerialNumber"]
KeyError: 'MetadataDevice'


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1393/FOTOS Camara-trampa 2
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1393/FOTOS Cámara-trampa
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1399 LAS PIEDRAS/Camara-Trampa-T4/FOTOS
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1399 LAS PIEDRAS/Camara-Trampa-T3/FOTOS
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1399 LAS PIEDRAS/Camara-Trampa-T1/FOTOS
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09048480
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09048888/FOTOS
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09049504/FOTOS
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1399 LAS PIEDRAS/Camara-Trampa-T4/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1399 LAS PIEDRAS/Camara-Trampa-T3/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1399 LAS PIEDRAS/Camara-Trampa-T1/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09048888/VIDEO


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09049504/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09049482/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1396 LA LEONA/HLPXGM09049093/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1392 NICTEHA 2/HLPXGM04049093/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1392 NICTEHA 2/HLPXGM09048480/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1392 NICTEHA 2/HLPXGM09048888/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1392 NICTEHA 2/HLPXGM09049482/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/NODO 3_13_0_1401 EL SANTUARIO/HLPXGM09048483/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/NODO 3_13_0_1401 EL SANTUARIO/HLPXGM09048507/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/NODO 3_13_0_1401 EL SANTUARIO/HLPXGM09048448/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/NODO 3_13_0_1401 EL SANTUARIO/HLPXGM09048449/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/NODO 3_13_0_1401 EL SANTUARIO/HLPXGM09048209/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1395 SANTA GERTRUDIS/HLPXGM09048888/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1395 SANTA GERTRUDIS/HLPXGM09049504/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1395 SANTA GERTRUDIS/HLPXGM09049482/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1395 SANTA GERTRUDIS/HLPXGM09049093/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1400 VILLAROSA/HLPXGM09048483/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1400 VILLAROSA/HLPXGM09048507/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1400 VILLAROSA/HLPXGM09048448/102RECNX/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1400 VILLAROSA/HLPXGM09048449/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1400 VILLAROSA/HLPXGM09048209/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1394 NICTEHA 1/HLPXGM09048480/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1394 NICTEHA 1/HLPXGM09048888/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1394 NICTEHA 1/HLPXGM09049504/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1394 NICTEHA 1/HLPXGM09049482/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_1_1394 NICTEHA 1/HLPXGM09049093/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1398 FRENO/HLPXGM09048483/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1398 FRENO/HLPXGM09048507/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1398 FRENO/HLPXGM09048448/VIDEO


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1398 FRENO/HLPXGM09048449/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM FOTOS/Nodo 3_13_0_1398 FRENO/HLPXGM09048209/VIDEOS


Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 248, in main
    extract_metadata_of_device_from_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 122, in extract_metadata_of_device_from_files
    f_pathlib = pathlib.Path(filename)
UnboundLocalError: local variable 'filename' referenced before assignment


# Checking result of `extract`

# 1)

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1394 NICTEHA 1/10262012` has no audios, removing from `/shared_volume/sipecam_subdirectories_02-06-2022.txt`

# 2)

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1394 NICTEHA 1/10262223` has no audios, removing from `/shared_volume/sipecam_subdirectories_02-06-2022.txt`

# 3)

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/found.000/dir0000.chk/HLPXGM 09048480` has no audios, removing from `/shared_volume/sipecam_subdirectories_02-06-2022.txt`

# 4)

**Executing without parallel feature**

In [5]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 254, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 168, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 147, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

**Problematic file:**

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_050000.WAV`

In [6]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_050000.WAV"
exiftool -a -j "$file"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_050000.WAV",
  "ExifToolVersion": 12.41,
  "FileName": "20210924_050000.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609",
  "FileSize": "5.5 MiB",
  "FileModifyDate": "2021:09:24 06:01:00-05:00",
  "FileAccessDate": "2022:06:05 04:38:25-05:00",
  "FileInodeChangeDate": "2022:04:27 17:28:22-05:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "File format error"
}]


CalledProcessError: Command 'b'file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_050000.WAV"\nexiftool -a -j "$file"\n'' returned non-zero exit status 1.

**Moving file to dir** `/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error`

In [7]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_050000.WAV"
dir_dst="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error"
mv "$file" "$dir_dst"

**Rerunning without parallel feature**

In [9]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 254, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 168, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 147, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

**Problematic file:**

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_053000.WAV`

In [10]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_053000.WAV"
exiftool -a -j "$file"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_053000.WAV",
  "ExifToolVersion": 12.41,
  "FileName": "20210924_053000.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609",
  "FileSize": "5.5 MiB",
  "FileModifyDate": "2021:09:24 06:31:00-05:00",
  "FileAccessDate": "2022:06:05 04:44:51-05:00",
  "FileInodeChangeDate": "2022:04:27 17:28:23-05:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "File format error"
}]


CalledProcessError: Command 'b'file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_053000.WAV"\nexiftool -a -j "$file"\n'' returned non-zero exit status 1.

**Moving file to dir** `/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error`

In [11]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_053000.WAV"
dir_dst="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error"
mv "$file" "$dir_dst"

**Rerunning without parallel feature**

In [13]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 254, in main
    extract_metadata_of_files(input_directory,
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 168, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/usr/local/lib/python3.8/dist-packages/simex/read_metadata_audio.py", line 147, in extract_date
    return get_date(comment_metadata)#example: 2021-07-28
  File "/usr/local/lib/python3.8/dist-packages/simex

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

**Problematic file:**

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_061000.WAV`

In [14]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_061000.WAV"
exiftool -a -j "$file"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_061000.WAV",
  "ExifToolVersion": 12.41,
  "FileName": "20210924_061000.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609",
  "FileSize": "5.5 MiB",
  "FileModifyDate": "2021:09:24 07:11:00-05:00",
  "FileAccessDate": "2022:06:05 04:55:53-05:00",
  "FileInodeChangeDate": "2022:04:27 17:28:23-05:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "File format error"
}]


CalledProcessError: Command 'b'file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210924_061000.WAV"\nexiftool -a -j "$file"\n'' returned non-zero exit status 1.

### As there are three files with the same problem this would be added into `check` of `simex` of `audio` cli files

In [17]:
%%bash
dir_audios="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error"
ls "$dir_audios"| wc -l

150


# check_if_there_are_audio_empty_files_in_dir_and_move_them

In [18]:
%%bash
SAVEIFS=$IFS;IFS=$(echo -en "\n\b")
for dir_to_be_processed in $(cat ~/list_dirs_with_audios.txt)
    do echo "$dir_to_be_processed"
    check_if_there_are_audio_empty_files_in_dir_and_move_them --input_directory_audio "$dir_to_be_processed" --number_of_processes 10
done
IFS=$SAVEIFS

/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Ultra/10263609
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Grabadoras/Audible 2/10261149
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_1_1393/Audible/10262012
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoths.N5.T4
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/Ultra.N5.T3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 1/AudioMoth
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 3
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPECAM AUDIOS/Nodo 3_13_0_1399 LAS PIEDRAS/Audiomoth 2/AudioMoths.N5.T4/Audio
/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/SIPEC

In [19]:
%%bash
dir_audios="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error"
ls "$dir_audios"| wc -l

154


**Rerunning without parallel feature**

In [21]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 11, in <module>
    load_entry_point('simex', 'console_scripts', 'extract_serial_numbers_dates_and_metadata_of_files_and_device')()
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 254, in main
    extract_metadata_of_files(input_directory,
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 168, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/read_

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

**Problematic file:**

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_104000.WAV`

In [22]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_104000.WAV"
exiftool -a -j "$file"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_104000.WAV",
  "ExifToolVersion": 12.41,
  "FileName": "20210923_104000.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609",
  "FileSize": "5.5 MiB",
  "FileModifyDate": "2021:09:23 11:41:00-05:00",
  "FileAccessDate": "2022:06:05 11:34:50-05:00",
  "FileInodeChangeDate": "2022:04:27 17:28:01-05:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "First 2.7 MiB of file is binary 0xff's"
}]


CalledProcessError: Command 'b'file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_104000.WAV"\nexiftool -a -j "$file"\n'' returned non-zero exit status 1.

**Moving file to dir** `/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error`

In [23]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_104000.WAV"
dir_dst="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error"
mv "$file" "$dir_dst"

**Rerunning without parallel feature**

In [25]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"

Traceback (most recent call last):
  File "/usr/local/bin/extract_serial_numbers_dates_and_metadata_of_files_and_device", line 11, in <module>
    load_entry_point('simex', 'console_scripts', 'extract_serial_numbers_dates_and_metadata_of_files_and_device')()
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 254, in main
    extract_metadata_of_files(input_directory,
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 168, in extract_metadata_of_files
    tuple_date, tuple_metadata_of_file = extract_metadata(filename)
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/extract_serial_numbers_dates_and_metadata_of_files_and_device.py", line 27, in extract_metadata
    date_of_file = read_metadata_audio.extract_date(filename)
  File "/LUSTRE/sacmod/SIPECAM/sipecam-metadata-extractor/src/simex/read_

CalledProcessError: Command 'b'dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"\nextract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed"\n'' returned non-zero exit status 1.

**Problematic file:**

`/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_103000.WAV`

In [26]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_103000.WAV"
exiftool -a -j "$file"

[{
  "SourceFile": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_103000.WAV",
  "ExifToolVersion": 12.41,
  "FileName": "20210923_103000.WAV",
  "Directory": "/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609",
  "FileSize": "5.5 MiB",
  "FileModifyDate": "2021:09:23 11:31:00-05:00",
  "FileAccessDate": "2022:06:05 11:49:55-05:00",
  "FileInodeChangeDate": "2022:04:27 17:28:01-05:00",
  "FilePermissions": "-rwxrwxrwx",
  "Error": "Entire file is binary 0xff's"
}]


CalledProcessError: Command 'b'file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_103000.WAV"\nexiftool -a -j "$file"\n'' returned non-zero exit status 1.

**Moving file to dir** `/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error`

In [27]:
%%bash
file="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/20210923_103000.WAV"
dir_dst="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/error"
mv "$file" "$dir_dst"

**Rerunning without parallel feature**

In [28]:
%%bash
dir_to_be_processed="/LUSTRE/sacmod/SIPECAM/cumulo_13/entrega_27.abril.2022/RESPALDO MICRO SD AURA/NODO 3_13_1_1393/GRABADORAS/ultra/10263609/10263609/"
extract_serial_numbers_dates_and_metadata_of_files_and_device --input_dir "$dir_to_be_processed" --parallel --number_of_processes 10


# Delivery /LUSTRE/sacmod/SIPECAM/Cumulo_96
